### PR TITLE
Additional SMTP email features

### PR DESCRIPTION
**Pull request by kpfleming**
* 'to' can now accept a list of addresses

* 'cc' and 'bcc' added with their usual meanings

* 'headers' can provide a list of custom headers

* 'X-Journal-Brief-Version' header always added

### CODE REVIEW

**Comment by kpfleming** [comment]
I'm going to investigate adding support for most (or all) of the same features for command-based delivery, but there were easier to add for SMTP delivery first :)

File: README.md [hunk 1]
```diff
@@ -207,3 +214,7 @@ secure the connection to the SMTP server
 
 * `password`: password to be used to authenticate to the SMTP server (only
 used if `user` is specified)
+
+* 'headers': dictionary of string keys and string values to be added as
+custom headers; if 'From', 'To', 'Cc', or 'Bcc' are included here they
+will override the corresponding configuration keys above
```
Comment (twaugh) [inline README.md:220]: Or maybe we should disallow those accounted for in other keys?

Comment (kpfleming) [reply README.md:220]: Yes, that's probably safer. Will add that.

**Comment by kpfleming** [comment]
Yes, I suspect that I'll end up changing this so that command-based delivery requires something like `mail -t`, otherwise the code would have to be able to insert recipient addresses into the supplied command string, and that will be complicated.

File: journal_brief/config.py [hunk 2]
```diff
@@ -292,6 +317,28 @@ def validate_email(self):
                 yield SemanticError('expected string', 'password',
                                     {'smtp': {'password': smtp['password']}})
 
+            if 'headers' in smtp:
+                    if not isinstance(smtp['headers'], dict):
+                        yield SemanticError('expected dict', 'headers',
+                                            {'smtp': {'headers': smtp['headers']}})
+
+                    if 'From' in smtp['headers']:
```
Comment (twaugh) [inline journal_brief/config.py:325]: What if it's `'from'`?

Comment (kpfleming) [reply journal_brief/config.py:325]: Excellent point; I had forgotten that SMTP headers are not case sensitive. Fix on the way.

**Comment by twaugh** [comment]
Thanks!

**Pull request merged**